### ISSUE TITLE
runwasi support

### ISSUE THREAD

**Issue by lum1n0us** [bytecodealliance/wasm-micro-runtime#2127]
## Background

WAMR is trying to be another Wasm runtime managed by containerd. It seems [runwasi](https://github.com/containerd/runwasi) is convenient.

Runwasi is a Rust project. The integration of *runwasi + wamr* includes two parts

- Creating rust binding for Wamr. The modification is in repo of [wasm-micro-runtime](https://github.com/bytecodealliance/wasm-micro-runtime/is)
- Integration in *runwasi*. The modification is in repo of [runwasi](https://github.com/containerd/runwasi)

This issue is used to track the first one. But it is open for every integration issues too.

### Minimum functionality requirement

> There are two modes of operation supported:
> - "Normal" mode where there is 1 shim process per container or k8s pod.
> - "Shared" mode where there is a single manager service running all shims in process.
> In either case you need to implement the `trait Instance`.

Summarize both implementations of [wasmtime](https://github.com/containerd/runwasi/blob/main/crates/containerd-shim-wasmtime/src/instance.rs) and [wasmedge](https://github.com/containerd/runwasi/blob/main/crates/containerd-shim-wasmedge/src/instance.rs):

- `new()` is about preparation. It need nothing from the wasm runtime.
- `start()` requires Wasm runtime loading and instantiation
- `kill()` handles signals. After searching, signals include *SIGKILL*, *SIGTERM*, *SIGINT* and *SIGHUP*. Basically, it requires Wasm runtime to stop or hang up the running instance
- `delete()` and `wait()` are about cleaning.

In a nutshell, the minimum requirement is to start the Wasm runtime(loading and instantiation) and to stop the wasm runtime.

### Thread safe and memory safe

Not sure this is the basic requirement, but after reading the [Foreign Function Interface of Rust](https://doc.rust-lang.org/nomicon/ffi.html#foreign-function-interface), bindings should be thread-safe and memory-safe. Ownership of resources transferred between in Rust and C , e.g. module file content and `wasm_trap_t`, should be taken care of.

### More

*TBD*
